app/src/features/surveys/observations/sampling-sites/SamplingSiteList.tsx

```diff
@@ -1,181 +1,195 @@
 import Button from '@mui/material/Button';
 import { grey } from '@mui/material/colors';
-import IconButton from '@mui/material/IconButton';
-import List from '@mui/material/List';
-import ListItem from '@mui/material/ListItem';
-import ListItemText from '@mui/material/ListItemText';
 import Toolbar from '@mui/material/Toolbar';
 import Typography from '@mui/material/Typography';
+import { CodesContext } from 'contexts/codesContext';
+import { SurveyContext } from 'contexts/surveyContext';
+import { useContext, useEffect, useState } from 'react';
 import { useHistory } from 'react-router';
+import { getCodesName } from 'utils/Utils';
 
 const SamplingSiteList = () => {
   const history = useHistory();
+  const surveyContext = useContext(SurveyContext);
+  const codesContext = useContext(CodesContext);
+  useEffect(() => {
+    codesContext.codesDataLoader.load();
+  }, [codesContext.codesDataLoader]);
+
+  const [anchorEl, setAnchorEl] = useState<MenuProps['anchorEl']>(null);
+  const [selectedSampleSiteId, setSelectedSampleSiteId] = useState<number | undefined>();
+
+  const handleMenuClick = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>, sample_site_id: number) => {
+    setAnchorEl(event.currentTarget);
+    setSelectedSampleSiteId(sample_site_id);
+  };
+
+  if (
+    !surveyContext.sampleSiteDataLoader.data ||
+    (surveyContext.sampleSiteDataLoader.isLoading && !codesContext.codesDataLoader.data) ||
+    codesContext.codesDataLoader.isLoading
+  ) {
+    return <CircularProgress color="inherit" />;
+  }
 
   return (
-    <Box display="flex" flexDirection="column" height="100%">
-      <Toolbar
-        sx={{
-          flex: '0 0 auto',
-          borderBottom: '1px solid #ccc'
+    <>
+      <Menu
+        open={Boolean(anchorEl)}
+        onClose={() => setAnchorEl(null)}
+        anchorEl={anchorEl}
+        anchorOrigin={{
+          vertical: 'top',
+          horizontal: 'right'
+        }}
+        transformOrigin={{
+          vertical: 'top',
+          horizontal: 'right'
         }}>
-        <Typography
-          sx={{
-            flexGrow: '1'
-          }}>
-          <strong>Sampling Sites</strong>
-        </Typography>
-        <Button
-          sx={{
-            mr: -1
-          }}
-          variant="contained"
-          color="primary"
+        <MenuItem
           onClick={() => {
-            history.push('sampling');
-          }}
-          startIcon={<Icon path={mdiPlus} size={1} />}>
-          Add
-        </Button>
-      </Toolbar>
-      <Box
-        position="relative"
-        display="flex"
-        flex="1 1 auto"
-        sx={{
-          overflowY: 'scroll',
-          background: grey[50],
-          '& .MuiAccordion-root + .MuiAccordion-root': {
-            borderTopStyle: 'solid',
-            borderTopWidth: '1px',
-            borderTopColor: grey[300]
-          }
-        }}>
-        <Box
+            history.push(`sampling/${selectedSampleSiteId}/edit`);
```

Comment:
Nit: avoid `history.push` in `onClick` event handlers and prefer using `component={RouterLink}`.

```javascript
import MenuButton from '@mui/...';
import { Link as RouterLink } from 'react-router-dom';

export default Example = () => {
  return (
    <MenuItem component={RouterLink} to={`sampling/${selectedSampleSiteId}/edit`}>
      ...
    </MenuItem>
  );
}
```

This isn't a major issue, but exposing the link to the browser is a generally a good idea because
1. a11y/semantics;
2. users can preview where the link will take them before they go, and;
3. users get access to the appropriate context menu, so they can do things like _Open in new tab_, etc.